File: lecture_06_python/problem_set/sentimental_cash/cash.py
from cs50 import get_float

quarter = 0.25
dime = 0.10
nickel = 0.05
penny = 0.01


def main():
    while True:
        cash_owend = get_float("Change: ")
        if cash_owend > 0:
            break

    result = get_number_of_coins(cash_owend)
    print(result)


def get_number_of_coins(owed):
    n = 0

    while owed >= quarter:
        n += 1
        owed = round(owed - quarter, 2)

    while owed >= dime:
        n += 1
        owed = round(owed - dime, 2)

    while owed >= nickel:
        n += 1
        owed = round(owed - nickel, 2)

    while owed >= penny:
        n += 1
        owed = round(owed - penny, 2)

    return n


if __name__ == "__main__":
    main()

File: lecture_01_c/problem_set/cash.c
#include <cs50.h>
#include <stdio.h>

const int quarter = 25;
const int dime = 10;
const int nickel = 5;
const int penny = 1;

int get_number_of_coins(int owed);

int main(void)
{
    int cash_owed;

    do
    {
        cash_owed = get_int("Cash owed: ");
    } while (cash_owed < 0);

    int result = get_number_of_coins(cash_owed);

    printf("%i\n", result);
}

int get_number_of_coins(int owed)
{
    int n = 0;
    while (owed >= quarter)
    {
        n++;
        owed -= quarter;
    }
    while (owed >= dime)
    {
        n++;
        owed -= dime;
    }
    while (owed >= nickel)
    {
        n++;
        owed -= nickel;
    }
    while (owed >= penny)
    {
        n++;
        owed -= penny;
    }
    return n;
}
